File: Interfaces/InventoryCountInterfaces/IIcHeaderService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface IICHeaderService
    {
        Task<ApiResponse<IEnumerable<ICHeaderDto>>> GetAllAsync();
        Task<ApiResponse<ICHeaderDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<ICHeaderDto>>> GetActiveAsync();
        Task<ApiResponse<ICHeaderDto>> CreateAsync(CreateICHeaderDto createDto);
        Task<ApiResponse<ICHeaderDto>> UpdateAsync(long id, UpdateICHeaderDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}

File: WebApi/Interfaces/InventoryCountInterfaces/IIcHeaderService.ts
import type { CreateIcHeaderDto, IcHeaderDto, UpdateIcHeaderDto } from '../../Models/index';
import type { ApiResponse, PagedResponse } from '../../ApiResponse';

export interface IIcHeaderService {
  getAll(): Promise<ApiResponse<IcHeaderDto[]>>;
  getById(id: number): Promise<ApiResponse<IcHeaderDto>>;
  create(createDto: CreateIcHeaderDto): Promise<ApiResponse<IcHeaderDto>>;
  update(id: number, updateDto: UpdateIcHeaderDto): Promise<ApiResponse<IcHeaderDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}
